Source/Core/Common/TraversalServer.cpp

```diff
@@ -160,28 +161,20 @@ static sockaddr_in6 MakeSinAddr(const TraversalInetAddress& addr)
   return result;
 }
 
-static void GetRandomBytes(void* output, size_t size)
+static void GetRandomBytes(unsigned char* output, size_t size)
+
 {
-  static u8 bytes[8192];
-  static size_t bytesLeft = 0;
-  if (bytesLeft < size)
-  {
-    ssize_t rv = read(urandomFd, bytes, sizeof(bytes));
-    if (rv != sizeof(bytes))
-    {
-      perror("read from /dev/urandom");
-      exit(1);
-    }
-    bytesLeft = sizeof(bytes);
-  }
-  memcpy(output, bytes + (bytesLeft -= size), size);
+  static std::independent_bits_engine<std::default_random_engine, sizeof(unsigned char) * 8,
```

Comment:
Isn't this almost always weaker than `std::random_device` ?

Reply:
`std::random_device` might be cryptographically secure on common implementations (i.e. if it's a wrapper around /dev/urandom), but that doesn't seem to be needed here. The recommended usage is to get a seed from the hardware and use that to run a PRNG (which is probably also faster than making a syscall every time).